rust/core-lib/src/event_context.rs

```diff
@@ -137,8 +137,12 @@ impl<'a> EventContext<'a> {
         let result = match cmd {
             Cut => Ok(self.with_editor(|ed, view, _| ed.do_cut(view))),
             Copy => Ok(self.with_editor(|ed, view, _| ed.do_copy(view))),
-            Find { chars, case_sensitive } => Ok(self.with_view(
-                |view, text| view.do_find(text, chars, case_sensitive))),
+            Find { chars, case_sensitive } =>
```

Comment:
Is the thinking here that existing clients will continue to work, but new clients will get the new behaviour?

One change with going async is that this should become a JSON-RPC 'notification', which means that the client should no longer be sending an `id` field and no longer expecting a response.

This comes down to how worried we are about breaking existing clients. I don't think we should be _too_ worried, but it's nice to be a _little_ worried.

Can we introduce the new behaviour with a new name? If we want we can even call it `find2` for the time being, and then we can return an error from the current method. Later on we could change `find` to point to the new method, keeping `find2` as an alias, and then get rid of that.

Or we can just put up an issue warning clients that find is about to break, which might be the better option.

Reply:
I would also prefer the last option